### PR TITLE
Remove AIRFLOW_V_2_8_PLUS for provider's compatibility tests

### PR DESCRIPTION
**Pull request by potiuk**
The AIRFLOW_V_2_8_PLUS is not needed as min airflow version already is 2.8.0+.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Extracted from #43556

File: contributing-docs/testing/unit_tests.rst [hunk 1]
```diff
@@ -1173,10 +1173,10 @@ are not part of the public API. We deal with it in one of the following ways:
 
 .. code-block:: python
 
-  from tests_common.test_utils.compat import AIRFLOW_V_2_8_PLUS
+  from tests_common.test_utils.compat import AIRFLOW_V_2_9_PLUS
 
 
-  @pytest.mark.skipif(not AIRFLOW_V_2_8_PLUS, reason="The tests should be skipped for Airflow < 2.8")
+  @pytest.mark.skipif(not AIRFLOW_V_2_9_PLUS, reason="The tests should be skipped for Airflow < 2.9")
   def some_test_that_only_works_for_airflow_2_8_plus():
```
Comment (vincbeck) [inline contributing-docs/testing/unit_tests.rst]:
```suggestion
  def some_test_that_only_works_for_airflow_2_9_plus():
```

File: providers/tests/openlineage/plugins/test_utils.py [hunk 2]
```diff
@@ -428,7 +427,7 @@ def test_serialize_timetable_2_9():
 
 
 @pytest.mark.skipif(
-    not AIRFLOW_V_2_8_PLUS or AIRFLOW_V_2_9_PLUS,
+    AIRFLOW_V_2_9_PLUS,
     reason="This test checks serialization only in 2.8 conditions",
 )
 def test_serialize_timetable_2_8():
```
Comment (vincbeck) [inline providers/tests/openlineage/plugins/test_utils.py:433]:
```suggestion
    reason="This test checks serialization only in 2.9 conditions",
)
def test_serialize_timetable_2_9():
```

Comment (potiuk) [reply providers/tests/openlineage/plugins/test_utils.py:433]: No - that's ok This test is run in 2.8 only. It has `skipif v2-9-plus` - so it only runs for 2.8 now - and we should remove it when we set min-airflow==2.9.0

Comment (potiuk) [reply providers/tests/openlineage/plugins/test_utils.py:433]: Previousely the condition was skipif (not AIRFLOW_V_2_8_PLUS or AIRFLOW_V_2_9_PLUS) = run only in 2.8. Yes. I know. Not obvious :)

Comment (potiuk) [reply providers/tests/openlineage/plugins/test_utils.py:433]: Maybe we should change the conditions to AIRFLOW_LESS_THAN_2_8 ... then probably some of the conditions would be easier to read :)

Comment (potiuk) [reply providers/tests/openlineage/plugins/test_utils.py:433]: (and I am not joking)

Comment (vincbeck) [reply providers/tests/openlineage/plugins/test_utils.py:433]: Oh I see, my bad. I think it is good enough as is, I just read it too quickly :)

**Review by vincbeck — APPROVED** [review]

**Review by o-nikolas — APPROVED** [review]

**Review by gopidesupavan — APPROVED** [review]

**Comment by potiuk** [comment]
Failing tests should be fixed after https://github.com/apache/airflow/pull/43771 is merged.

**Pull request merged**